jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -52,10 +56,14 @@
    * @param sourceFile the source file to add to the layer
    * @param extractionPath the path in the container file system corresponding to the {@code
    *     sourceFile}
+   * @param permissions the file permissions on the container. Use {@code null} to use defaults (644
+   *     for files, 755 for directories)
    */
-  public LayerEntry(Path sourceFile, AbsoluteUnixPath extractionPath) {
+  public LayerEntry(
```

Comment:
We might want to consider in this or a follow up PR: but we seem to now have a lot of code that is LayerEntry(file, path, null) where maybe static initializers could be a nice entry point? My naming is a little off, but I dunno.
```
LayerEntry.asEntry(sourceFile, extractionPath)
LayerEntry.asEntryWithPermissions(sourceFile, extractionPath, permissions)
```
and this constructor with optional args then becomes private/package private

Reply:
But those a lot of `new LayerEntry(file, path, null)` are only in testing. I vaguely remember the reason we did not add the 2-arg version was that internally there will be only a couple places for instantiating `LayerEntry`? (I may be wrong.)

Reply:
Yeah, `LayerEntry` isn't intended to be a user-facing class, so I don't think this matters that much at the moment.

Reply:
Oh, well if that's the case, lets just leave this as is.

Reply:
Filed an issue in case we want to go back to it later: https://github.com/GoogleContainerTools/jib/issues/1168